api/src/main/java/org/openmrs/module/fhir2/api/impl/FhirConditionServiceImpl.java

```diff
@@ -57,20 +47,14 @@ public class FhirConditionServiceImpl extends BaseFhirService<Condition, Obs> im
 	private SearchQuery<org.openmrs.Obs, Condition, FhirConditionDao<org.openmrs.Obs>, ConditionTranslator<org.openmrs.Obs>, SearchQueryInclude<Condition>> searchQuery;
 	
 	@Override
-	public IBundleProvider searchConditions(ReferenceAndListParam patientParam, TokenAndListParam code,
-	        TokenAndListParam clinicalStatus, DateRangeParam onsetDate, QuantityAndListParam onsetAge,
-	        DateRangeParam recordedDate, TokenAndListParam id, DateRangeParam lastUpdated, @Sort SortSpec sort,
-	        HashSet<Include> includes) {
+	public IBundleProvider searchConditions(ConditionSearchParams conditionSearchParams) {
+		if (conditionSearchParams.getClinicalStatus() != null
+		        && !conditionSearchParams.getClinicalStatus().getValuesAsQueryTokens().isEmpty()) {
+			throw new IllegalArgumentException("The clinicalStatus parameter is not supported in this search.");
+		}
```

Comment:
@ibacher i added the check to ensure that from versions lower than 2.2.0, clinicalStatus is not required. Any thoughts?